### ISSUE TITLE
Arc: implement in-app messaging

### ISSUE THREAD

**Issue by altsang** [strongloop/strong-arc#1290]
As an Arc user I would want the ability to send and receive messages within Arc itself so that I don't have to go to another client to communicate back to SL.

- [x] using Intercom  in-app messaging
- [x] conduct UX on insertion point to invoke functionality - we we use a separate button/widget or menu item on the hamburger icon?
- [x] Intercom  messaging widget needs to be stylzed for Arc - see http://docs.intercom.io/configuring-Intercom/in-app-messenger for more info

/cc @sumitha

**Comment by anthonyettinger** [comment]
@stacy-hunt do we have a mock?

**Comment by altsang** [comment]
on the agenda to discuss today

**Comment by chandadharap** [comment]
@altsang Anthony is blocked.

**Comment by stacy-hunt** [comment]
Proposed Design: Intercom icon is slightly customised to use Strongloop colours. Note that it has a subtle dropshadow:

![image](https://cloud.githubusercontent.com/assets/1924932/8415232/30fe85b2-1e52-11e5-9018-d4ab140b3971.png)

https://drive.google.com/open?id=0B1FTzkB7697GX2NKZWFGdUJMMEU

Link to usage examples: https://drive.google.com/open?id=0B1FTzkB7697GdV9TWlVicWNzUW8

Link to svg for icon: https://drive.google.com/open?id=0B1FTzkB7697GX2NKZWFGdUJMMEU

**Comment by altsang** [comment]
thank @stacy-hunt - can we use the above as a last resort?  we thikn we don't need to use the Intercom default icon - Waffle uses their own (?) icon.  @anthonyettinger is currently looking at adding a menu item in the hamburger icon and also see if we can create our own icon that invokes their widget - the messaging widget is the piece where they'll offer limited customization through CSS

**Comment by altsang** [comment]
unblocked Anthony on getting started with segment and intercom

**Comment by stacy-hunt** [comment]
Yeah, if Anthony's unblocked then we can discuss further requirements and
iterate on it if necessary in the UX meeting.

**Comment by anthonyettinger** [comment]
@stacy-hunt -- you were going to update the SVG icon to use in the navbar.

**Comment by stacy-hunt** [comment]
Here's a link to the folder that contains the navbar icons:

https://drive.google.com/folderview?id=0B1FTzkB7697GaFl6QnlFUDh3Z2c&usp=sharing

Here's a screenshot of what the new icon should look like in context:
![image](https://cloud.githubusercontent.com/assets/1924932/8559967/2e9636e6-2567-11e5-8f78-0b5654d40c3d.png)

**Comment by anthonyettinger** [comment]
@altsang you can verify it on this branch. I will hold off merging to master until you give the OK. We will need to flip a few switches when that happens.

https://github.com/strongloop/strong-arc/tree/feature/intercom-integration

switch to this branch then run `gulp build`

**Comment by altsang** [comment]
reviewed on `feature/intercom-integration`, can be merged to master, but please hide behind a feature flag, I will coordinate with Sumitha on customer communication

**Comment by anthonyettinger** [comment]
@altsang we still need to switch it out of test environment before we merge to master. re-opening.

**Comment by altsang** [comment]
ahh yes - good catch

**Comment by anthonyettinger** [comment]
can we get a ticket for putting behind a feature flag and changing to production env?

**Comment by altsang** [comment]
oops - I created a duplicate - https://github.com/strongloop-internal/scrum-loopback/issues/421, do you want to #1479? I have an action item I'll check off for Sumitha coordination, thx!

**Comment by anthonyettinger** [comment]
#1479 is good. I put it in Planning
